Week1/homework/app.js

```diff
@@ -1,11 +1,153 @@
+    masnavi:
+      'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1171037882i/83264._SY475_.jpg',
+    black_holes_baby_universes:
+      'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1386924312i/53200.jpg',
+    blindness:
+      'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1327866409i/2526.jpg',
+    one_Hundred_years_solitude:
+      'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1327881361i/320.jpg',
+    girl_train:
+      'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1490903702i/22557272.jpg',
+    habits_highly_effective_people:
+      'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1421842784i/36072.jpg',
+    philosophy_fear:
+      'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1328701163i/5800070.jpg',
+    sapiens:
+      'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1420585954i/23692271.jpg',
+    lust_life:
+      'https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1348532272i/79834.jpg',
+  };
+
+  function makeList() {
+    const body = window.document.getElementById('body');
+    const ul = window.document.createElement('ul');
+    // I think there is no need to add id to lists, because we can find it just by item name at first I used this codes but now I comment it:
+    // li.setAttribute('id', item);
+    // console.log(bookCover[item]);
+
+    for (const item in bookInfo) {
+      if (item) {
+        const li = window.document.createElement('li');
+        const image = window.document.createElement('img');
+        image.setAttribute('src', bookCover[item]);
+        li.appendChild(image);
+
+        ul.appendChild(li);
+        const h3Title = window.document.createElement('h3');
+        h3Title.innerHTML = bookInfo[item].title;
+        li.appendChild(h3Title);
+
+        const bookProperties = Object.keys(bookInfo[item]);
+        for (const itemChild in bookProperties) {
+          if (itemChild !== 0) {
+            const bookProperty = bookProperties[itemChild];
+            const h4Author = window.document.createElement('h4');
+            h4Author.innerHTML = bookProperty + ':' + bookInfo[item][bookProperty];
+            li.appendChild(h4Author);
+          }
+        }
+        // I made another loop inside each object to get out each property inside nested object insted of calling them by their names like this command codes:
+        // const h4Author = window.document.createElement('h4');
+        // h4Author.innerHTML = "Author:" + bookInfo[item].author;
+        // li.appendChild(h4Author);
+
+        // const h4Language = window.document.createElement('h4');
+        // h4Language.innerHTML = "Language:" + bookInfo[item].language;
+        // li.appendChild(h4Language);
+      }
+    }
+    body.appendChild(ul);
+  }
+
+  window.onload = () => makeList();
```

Comment:
If you remember from our lesson of Week1, if we make use of event listeners like this i.e `window.onload` instead of using `addEventListeners`, it overrides any previous listeners. For example, you have an event listener here as well as on line 32. Since this one is parsed at the end by JavaScript, it will override the previous event listener. I hope you were aware of it when you did it like this 🙂